tests/test-stake-pool-sdk.ts

```diff
@@ -178,6 +180,81 @@ describe("test basic functionalities", () => {
       );
     });
 
+    it("it calcSolDeposit() and calcSolDepositsInverse() works correctly", async () => {
+      const socean = new Socean("testnet");
+      const stakePool = await socean.getStakePoolAccount();
+
+      const depositAmountSol = Math.random() / 4;
+      const depositAmount = Math.round(depositAmountSol * LAMPORTS_PER_SOL);
+      const depositAmountLamports = new Numberu64(depositAmount);
+      const { dropletsReceived, dropletsFeePaid, referralFeePaid } =
+        calcSolDeposit(depositAmountLamports, stakePool.account.data);
+
+      const {
+        lamportsStaked,
+        dropletsFeePaid: inverseDropletsFeePaid,
+        referralFeePaid: inverseReferralFeePaid,
+      } = calcSolDepositInverse(dropletsReceived, stakePool.account.data);
+
+      expect(
+        depositAmountLamports.toNumber() - lamportsStaked.toNumber(),
+      ).to.be.at.most(2);
+      expect(dropletsFeePaid.toNumber()).to.eq(
+        inverseDropletsFeePaid.toNumber(),
+      );
+      expect(referralFeePaid.toNumber()).to.eq(
+        inverseReferralFeePaid.toNumber(),
+      );
+    });
+
+    it("it calcWithdrawals() and calcWithdrawalsInverse() works correctly", async () => {
+      const socean = new Socean("testnet");
+      const stakePool = await socean.getStakePoolAccount();
+      const validatorList = await socean.getValidatorListAccount(
+        stakePool.account.data.validatorList,
+      );
+      const scnSolAtaAcctInfo = await scnSolToken.getAccountInfo(
+        scnSolAtaPubkey,
+      );
+      const initialBalanceDroplets = scnSolAtaAcctInfo.amount;
+
+      const withdrawAmountDroplets = new Numberu64(
+        Math.round(Math.random() * initialBalanceDroplets.toNumber()),
+      );
+
+      const validatorWithdrawalReceipts = await calcWithdrawals(
+        withdrawAmountDroplets,
+        stakePool,
+        validatorList.account.data,
+      );
+
+      let totalLamportsReceived: Numberu64 = new Numberu64(0);
+      validatorWithdrawalReceipts.forEach((receipt) => {
+        totalLamportsReceived = Numberu64.cloneFromBN(
+          totalLamportsReceived.add(receipt.withdrawalReceipt.lamportsReceived),
+        );
+      });
+
+      const validatorWithdrawalReceiptsInverse = await calcWithdrawalsInverse(
+        totalLamportsReceived,
+        stakePool,
+        validatorList.account.data,
+      );
+
+      let totalDropletsToUnstake: Numberu64 = new Numberu64(0);
+      validatorWithdrawalReceiptsInverse.forEach((receipt) => {
+        totalDropletsToUnstake = Numberu64.cloneFromBN(
+          totalDropletsToUnstake.add(
+            receipt.withdrawalReceipt.dropletsUnstaked,
+          ),
+        );
+      });
+
+      expect(totalDropletsToUnstake.toNumber()).to.eq(
+        withdrawAmountDroplets.toNumber(),
+      );
```

Comment:
i just changed the testnet stake pool to have withdrawal fees, will take effect on the next epoch. In the meantime, you might wanna change this to test using mainnet parameters where there are withdrawal fees. This might cause the result to differ by 2 lamports like in deposit.